flink-end-to-end-tests/test-scripts/test_kubernetes_materialized_table.sh

```diff
@@ -0,0 +1,242 @@
+    http://localhost:$SQL_GATEWAY_REST_PORT/sessions/$session_handle/statements | jq -r '.operationHandle')
+  get_operation_result $session_handle $operation_handle
+}
+
+function create_filesystem_source() {
+  local session_handle=$1
+  local table_name=$2
+  local path=$3
+
+  create_source_result=$(curl -s -X POST \
+    -H "Content-Type: application/json" \
+    -d "{\"statement\": \"CREATE TABLE $table_name (\
+      \`timestamp\` TIMESTAMP_LTZ(3),\
+      \`user\` STRING,\
+      \`type\` STRING\
+    ) WITH (\
+      'format' = 'csv'\
+    )\"}" \
+    http://localhost:$SQL_GATEWAY_REST_PORT/sessions/$session_handle/statements/)
+
+    echo $create_source_result
+}
+
+S3_ENDPOINT=${S3_ENDPOINT//localhost/$(get_host_machine_address)}
+echo "[INFO] Create Materialized Table in Application Mode"
+session_options="{\"table.catalog-store.kind\": \"file\",
+                 \"execution.target\": \"kubernetes-application\",
+                 \"kubernetes.cluster-id\": \"${APPLICATION_CLUSTER_ID}\",
+                 \"kubernetes.container.image.ref\": \"${FLINK_IMAGE_NAME}\",
+                 \"table.catalog-store.file.path\": \"$MATERIALIZED_TABLE_DATA_DIR/\",
+                 \"kubernetes.rest-service.exposed.type\": \"NodePort\",
+                 \"s3.endpoint\": \"$S3_ENDPOINT\",
+                 \"workflow-scheduler.type\": \"embedded\"}"
+
+session_handle=$(open_session "$session_options")
+echo "[INFO] Session Handle $session_handle"
+
+mkdir -p $MATERIALIZED_TABLE_DATA_DIR/catalog
+configure_session "$session_handle" "create catalog if not exists test_catalog with (\
+                                                'type' = 'test-filesystem',\
+                                                'default-database' = 'test_db',\
+                                                'path' = '$MATERIALIZED_TABLE_DATA_DIR/'\
+                                    )"
+# prepare default db & default catalog
+configure_session $session_handle "use catalog test_catalog"
+configure_session $session_handle "create database if not exists test_db"
+create_filesystem_source $session_handle "filesystem_source"
+
+# 1. create materialized table in continuous mode
+create_materialized_table_in_continous_mode $session_handle "my_materialized_table_in_continuous_mode"
+
+configure_session $session_handle "set 'execution.checkpointing.savepoint-dir' = '$MATERIALIZED_TABLE_DATA_DIR/savepoint'"
+kubectl wait --for=condition=Available --timeout=30s deploy/${APPLICATION_CLUSTER_ID} || exit 1
+jm_pod_name=$(kubectl get pods --selector="app=${APPLICATION_CLUSTER_ID},component=jobmanager" -o jsonpath='{..metadata.name}')
+wait_num_checkpoints $jm_pod_name 1
+
+# 2. suspend & resume materialized table in continuous mode
+execute_statement $session_handle "alter materialized table my_materialized_table_in_continuous_mode suspend"
+
+kubectl delete deployment $APPLICATION_CLUSTER_ID
```

Comment:
Why do we need to manually delete clusters？

Reply:
In my test, I found that after "stop with Savepoint", the deployment does not end. I need to spend some time to verify it again. In theory, it should exit automatically.

Reply:
After my testing, it seems that the failure of deployment to exit normally is a bug in SqlDriver when splitting sql. When there is no semicolon at the end of the sql statement, it will enter an infinite loop and thus infinitely submit new jobs. Since SqlDriver needs time to fix this bug, currently we can add a semicolon by default when refreshing the statement of the job to avoid this problem.

Reply:
> After a discuss with @fsk119 , We decide to directly fix this is this pr, the commit is: [de3675e](https://github.com/apache/flink/commit/de3675e8a4a5db4c25e0ab8e15d58e31694af7b3)

BTW, the CI is failed.

Reply:
It seems there are some issues with executing DML statements, so I will switch to executing a simple DDL statement first.